CMakeLists.txt

```diff
@@ -45,6 +44,11 @@ option(OCOS_BUILD_PYTHON "Enable building the Python package" OFF)
 option(OCOS_BUILD_JAVA "Enable building the Java package" OFF)
 option(OCOS_BUILD_ANDROID "Enable building the Android package" OFF)
 
+# Pre/post processing operator collections.
+option(OCOS_ENABLE_PPP_VISION "Enable operators for adding pre/post processing to vision models" ON)
```

Comment:
The existing options are created for selected operator build since the 3rd party libraries binary take most of proportion of footprint. It meets the footprint requirements while avoiding #ifdef flood in the source code.

We can group these options to create the domain package, instead of adding of these new options which is not same granularity.

Reply:
I also don't want excessive `#ifdef`s in the code. 

If I look at the existing tokenizer options for example, OCOS_ENABLE_PPP_TEXT could simply turn on the required tokenizer options (e.g. turn on OCOS_ENABLE_BERT_TOKENIZER but turn off the GPT2 tokenizer as nobody is going to be running GPT2 on mobile) without adding any new ifdefs. i.e. all the handling for it is at the cmake level

For vision I saw there's an existing OPENCV option, but 'opencv' seems like an internal dependency/implementation detail. Can a user instead pick functionality (e.g. 'vision') and based on that we enable the required dependencies like opencv? 

Do the existing operators in the operators\cv2 have any production usage currently? I don't need any of them for the scenarios we're targeting.

Reply:
Adding a new CMake option criteria is whether the new one or some operators increase the footprint significantly. 
The new options you are adding here seem to be a higher-level conception, like option group, which is what I mean 'not same granularity'. 
We can create the group option in CMakeLists.txt technically, but it would be better to create some separated files to specify which options should be turned on for the certain domains. So, we still can keep a simple and straightforward cmake option design pattern.

Reply:
I don't care if the grouping is done in this file or some external file that specifies a bunch of cmake options. Assuming we do the latter, I can remove the OCOS_ENABLE_PPP_* values.

Not clear how I should structure it to provide a way to select the operators for vision pre/post processing though, which is just the 2 new ones. I don't want the existing ones in the cv2 folder that are currently enabled by OCOS_ENABLE_OPENCV. Can you please provide a suggestion of what you have in mind?

Reply:
Adding a new flag of OCOS_ENABLE_OPENCV_IMG or some other name? A OPENCV in the middle of the name means it depends on OPENCV library, actually there are already OCOS_ENABLE_OPEN_CODEC, we may reuse that one and make any change you want since they are not using in any product yet.